cvmfs/loader.cc

```diff
@@ -616,6 +618,9 @@ int FuseMain(int argc, char *argv[]) {
 
   int retval;
 
+  // check if already mounted in debug mode
+  debug_mode_ = getenv("__CVMFS_DEBUG_MODE__") != NULL;
```

Comment:
Moving this from line 673 to the front seems unnecessary. Let's try without.

Reply:
no we have to move it to the front because of line 
https://github.com/cvmfs/cvmfs/blob/1030e2e0a27c37bc0703acbafb9f34b1502c90b7/cvmfs/loader.cc#L640

Reply:
But if you look into the three lines before them, we overwrite `debug_mode_` anyway.

Reply:
you mean the string compare if `--debug` is set?

as far as i understood there are 2 cases how you can be in debug mode:
- you mounted in debug mode
- you set the debuglog in the client config

the `getenv` checks if we already are in debug mode and sets `debug_mode_` true in that case. and `--debug` sets `_debug_mode_` true if in the reload command it is requested to switch to debug mode.

in the old location it will definitely overwrite the request to change from non-debug to debug mode (as `__CVMFS_DEBUG_MODE__` is only set in `ManagedExec` in `posix.cc`

but again.. maybe i dont fully understand how everything interacts with each other.

Reply:
i think you are right with putting the line back to where it was. looking closer at the function this is a totally different code path and not part of the `reload` functionality